ward/_terminal.py

```diff
@@ -789,68 +798,149 @@ def get_source(self, err: TestFailure, test_result: TestResult) -> RenderableTyp
         return Padding(src, (1, 0, 1, 4))
 
     def get_pretty_comparison_failure(self, err: TestFailure) -> RenderableType:
-        if err.operator is Comparison.Equals:
-            return self.get_pretty_failure_for_equals(err)
-        elif err.operator in {Comparison.In, Comparison.NotIn}:
-            return self.get_pretty_failure_for_in(err)
-        else:
-            return Text("", end="")
-
-    def get_pretty_failure_for_equals(self, err: TestFailure) -> RenderableType:
-        diff_msg = Text.assemble(
-            ("LHS ", "pass.textonly"),
-            ("vs ", "default"),
-            ("RHS ", "fail.textonly"),
-            ("shown below", "default"),
-        )
-
-        diff = Diff(
-            err.lhs,
-            err.rhs,
-            width=self.terminal_size.width - 24,
-            show_symbols=self.show_diff_symbols,
+        parts = [
+            self.get_operands(err),
+            self.get_diff(err),
+            self.get_message(err),
+        ]
```

Comment:
I'm imagining this is where you could hook in to add new parts to the output.

One I had an idea for, but is too expansive to try to put in this PR - if you get a comparison failure, and the objects involved have custom implementations for that operator, you could go look up the source code for it and render it.

Reply:
Not 100% following this:

> if you get a comparison failure, and the objects involved have custom implementations for that operator, you could go look up the source code for it and render it.

Reply:
Woops, missed this comment - the idea here is that, if I'm doing a comparison with objects that have custom comparison operators (like an overridden `__eq__` or whatever), the first place I might want to look at to understand why a comparison failed might be that custom operator's implementation. So, we could display it's source code in an info panel under the diff. I like the idea that the output is extensible and customizable, so that more information and context about the failure can be packed into the output if desired.

Reply:
Ah I see what you mean now, thanks! That would be cool :)